packages/css-modules-flow-types-loader/index.js

```diff
@@ -3,12 +3,52 @@
 import fs from 'fs';
 import printFlowDefinition from 'css-modules-flow-types-printer';
 
+/**
+ * @desc
+ * Parse the string representation of css-modules ICSS source code
+ * and output local classNames if any were found
+ *
+ * @param {String} content
+ * @return {Object<String, String>}
+ */
 function getTokens(content) {
-  const cssTokens = content.replace(/\s/g, '').match('exports.locals=(.*);');
-  if (cssTokens) {
-    return JSON.parse(cssTokens[1]);
+  // Only `locals` export is desired
+  const locals = content.match(/exports\.locals = ([\s\S]*);/);
+
+  if (!locals) return {};
+
+  /**
+   * Collect object keys by parsing the string
+   * This is needed because external composes result in require calls concatenation,
+   * so we cannot simply JSON.parse (breaks on first + sign)
+   *
+   * @example
+   * // Following input css
+   * // .root {
+   * //     composes: foo from 'styles/bar.scss';
+   * //     color: red;
+   * // }
+   * // Will result in something like this pseudo-source (we get it as a string):
+   * exports.locals = {
+   *   "root": "Menu_root_1TJ" + require("-!css-loader!styles/bar.scss").locals["foo"] + "",
+   * };
+   */
+  const tokens = [];
```

Comment:
As the end result is an object, why not use that here instead of an array? Then you can also get rid of the `reduce` in the end?

Reply:
This pertains to the question I've posted in the issue thread ... you're using only the keys in the printer, so we might as well skip the object and pass along the array.

Here I'm faking the values anyway... however this is a decision that needs to be done on your side as you might have some plans for future version that I don't see and other code depending on the printer might need to be updated

Reply:
Only the keys are needed here, and I cannot see us using anything else in the printer. 

If you want to change the printer to take a list of keys then that is fine by me. : )